### PR TITLE
Add ai21 llm

### PR DESCRIPTION
**Pull request by D-coder22**
# Description

This PR adds support for [AI21 Labs](https://www.ai21.com/) LLMs. I have added unit tests for basic functionality. I have also included an example notebook and added that to the documentation. AI21 Labs Python SDK does not have a streaming functionality, hence I have not implemented it and if called it raises a NotImplementedError.

## Type of Change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Added new unit/integration tests
- [x] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama_index/llms/ai21.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+            **kwargs,
+        }
+
+    @llm_completion_callback()
+    def complete(self, prompt: str, **kwargs: Any) -> CompletionResponse:
+        all_kwargs = self._get_all_kwargs(**kwargs)
+
+        import ai21
+
+        ai21.api_key = self._api_key
```
Comment (Disiok) [inline llama_index/llms/ai21.py:102]: curious: why not set this in the constructor?

File: docs/examples/llm/ai21.ipynb
Comment (Disiok) [inline docs/examples/llm/ai21.ipynb:1]: Could you help clean up the error output in the notebook?

File: llama_index/llms/ai21.py (hunk 1 above)
Comment (D-coder22) [reply llama_index/llms/ai21.py:102]: I had set the api_key in the constructor initially. AI21's Python SDK sets it the api_key globally. Which gave me the issue where, say I created two AI21 LLM objects with different keys, both would end up using the key that the 2nd object was created with.

The current method of setting the api_key when the chat/complete call is made, solves that issue.

File: docs/examples/llm/ai21.ipynb
Comment (D-coder22) [reply docs/examples/llm/ai21.ipynb:1]: I had left the error in there to show that one could create two different LLM objects with two different keys. The 2nd Object used a `"Bad_key"` creating an error. I wasn't entirely sure what would be a better way of showing this capability. I can make the 2nd object use another valid `api_key` and show output for both calls.

Comment (Disiok) [reply docs/examples/llm/ai21.ipynb:1]: ah so it's intentional, gotcha. Perhaps it's okay then. I'll take a another peek.

**Pull request merged**